corehq/apps/userreports/views.py

```diff
@@ -654,6 +640,33 @@ def page_context(self):
             'report_builder_events': self.request.session.pop(REPORT_BUILDER_EVENTS_KEY, [])
         }
 
+    def _handle_exception(self, response, exception):
+        if self.existing_report and self.existing_report.report_meta.edited_manually:
+            error_message_base = _(
+                'It looks like this report was edited by hand and is no longer editable in report builder.'
+            )
+            if toggle_enabled(self.request, toggles.USER_CONFIGURABLE_REPORTS):
+                error_message = '{} {}'.format(error_message_base, _(
+                    'You can edit the report manually using the <a href="{}">advanced UI</a>.'
+                ).format(reverse(EditConfigReportView.urlname, args=[self.domain, self.existing_report._id])))
+            else:
+                error_message = '{} {}'.format(
+                    error_message_base,
+                    _('You can delete and recreate this report using the button below, or '
+                      'report an issue if you believe you are seeing this page in error.')
+                )
+            response['error_message'] = error_message
+            return response
+        elif isinstance(exception, DataSourceConfigurationNotFoundError):
```

Comment:
Yea not sure. I was thinking that it doesn't make much sense to show end users a message that says the datasource has been deleted. The action for them is correct (delete and recreate).